libs/pika/async_mpi/include/pika/async_mpi/mpi_polling.hpp

```diff
@@ -72,38 +92,169 @@ namespace pika::mpi::experimental {
         // Background progress function for MPI async operations
         // Checks for completed MPI_Requests and sets ready state in waiting receivers
         PIKA_EXPORT pika::threads::detail::polling_status poll();
+
+        // utility function to avoid duplication in eager check locations
+        PIKA_EXPORT bool poll_request(MPI_Request /*req*/);
+
+        // -----------------------------------------------------------------
+        using semaphore_type = pika::counting_semaphore<>;
+        //
+        PIKA_EXPORT std::shared_ptr<semaphore_type> get_semaphore(stream_type s);
+
+        inline constexpr bool throttling_enabled = false;
+
+        /// flags that control how mpi continuations are handled
+        enum class handler_mode : std::uint32_t
+        {
+            /// enable the use of a dedicated pool for polling mpi messages.
+            use_pool = 0x01,
+
+            /// this bit enables the inline invocation of the mpi request, when set
+            /// the calling thread performs the mpi operation, when unset, a transfer
+            /// is made so that the invocation happens on a new task that
+            /// would normally be on a dedicated pool if/when it exists
+            request_inline = 0x02,
+
+            /// this bit enables the inline execution of the completion handler for the
+            /// request, when unset a transfer is made to move the completion handler
+            /// from the polling thread onto a new one
+            completion_inline = 0x04,
+
+            /// this bit enables the use of a high priority task flag when a
+            /// completion is handled so that the continuation is executed quickly
+            /// if transfered to a new task, or resumed from sleep.
+            high_priority = 0x08,
+
+            /// 3 bits control the handler method,
+            method_mask = 0x70,
+
+            /// the individual methods that are supported for dispatching continuations
+            yield_while = 0x00,
+            suspend_resume = 0x10,
+            new_task = 0x20,
+            continuation = 0x30,
+            original = 0x40,
+
+            /// Default flags are to invoke inline, but transfer completion using a dedicated pool
+            default_mode = use_pool | request_inline | high_priority | new_task,
+        };
+
+        // 2 bits define continuation mode
+        inline handler_mode get_handler_mode(int flags)
+        {
+            return static_cast<handler_mode>(flags & to_underlying(handler_mode::method_mask));
+        }
+
+        // 1 bit defines pool or no pool
+        inline bool use_HP_com(int mode)
```

Comment:
Do you think we could expand `com` to `completion` here and elsewhere:
```suggestion
        inline bool use_HP_completion(int mode)
```
?
`com` is too close to `comm` in my opinion and we're not running out of characters.